extensions/core/deployment/src/test/java/io/quarkiverse/cxf/deployment/test/MockTest.java

```diff
@@ -0,0 +1,101 @@
+package io.quarkiverse.cxf.deployment.test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.inject.Singleton;
+import jakarta.jws.WebMethod;
+import jakarta.jws.WebService;
+
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.AbstractPhaseInterceptor;
+import org.apache.cxf.phase.Phase;
+import org.assertj.core.api.Assertions;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+import io.quarkiverse.cxf.annotation.CXFClient;
+import io.quarkiverse.cxf.annotation.CXFEndpoint;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class MockTest {
+
+    @RegisterExtension
+    public static final QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(HelloService.class, PropertiesCounterInterceptor.class))
+            .overrideConfigKey("quarkus.cxf.endpoint.\"/helloMockConfig\".in-interceptors", "#propertiesCounterInterceptor")
+            .overrideConfigKey("quarkus.cxf.client.helloMock.client-endpoint-url", "http://localhost:8081/services/helloMock")
+            .overrideConfigKey("quarkus.cxf.client.helloMock.service-interface", HelloService.class.getName())
+            .overrideConfigKey("quarkus.cxf.client.helloMockConfig.client-endpoint-url",
+                    "http://localhost:8081/services/helloMockConfig")
+            .overrideConfigKey("quarkus.cxf.client.helloMockConfig.service-interface", HelloService.class.getName());
+
+    @CXFClient("helloMock")
+    HelloService helloMockClient;
+
+    @CXFClient("helloMockConfig")
+    HelloService helloMockConfigClient;
+
+    @Inject
+    @Named("propertiesCounterInterceptor")
+    PropertiesCounterInterceptor interceptor;
+
+    @CXFEndpoint("/helloMock")
+    HelloService helloMockService() {
+        final HelloService result = Mockito.mock(HelloService.class);
+        Mockito.when(result.hello("Mock")).thenReturn("helloMock!");
+        return result;
+    }
```

Comment:
@famod this is the idea I had to support mocking endpoints in tests https://github.com/quarkiverse/quarkus-cxf/discussions/1217 Does it somehow match your needs?